sjsonnet/src/sjsonnet/Evaluator.scala

```diff
@@ -598,7 +598,7 @@ class Evaluator(resolver: CachedResolver,
           builder.put(k, v)
         }
     }
-    cachedObj = new Val.Obj(objPos, builder, false, if(asserts != null) assertions else null, sup)
+    cachedObj = new Val.Obj(objPos, Platform.compactHashMap(builder), false, if(asserts != null) assertions else null, sup)
```

Comment:
I don't remember all the different call sites and didn't look into each one of them for this review. Can we know the size up-front and do it more efficiently (i.e. construct the right type directly instead of compacting later)? At least the type should be known in all cases.

Reply:
Makes sense! I"m doing that in the followup - https://github.com/databricks/sjsonnet/pull/197 that replaces/subsumes the work from this PR.